nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/state/flow/flow.actions.ts

```diff
@@ -139,262 +156,348 @@ export const loadRemoteProcessGroupSuccess = createAction(
+export const navigateWithoutTransform = createAction(
+    `${CANVAS_PREFIX} Navigate Without Transform`,
+    props<{ url: string[] }>()
+);
 
 /*
     Palette actions
  */
 
 export const setNavigationCollapsed = createAction(
-    '[Canvas] Set Navigation Collapsed',
+    `${CANVAS_PREFIX} Set Navigation Collapsed`,
     props<{ navigationCollapsed: boolean }>()
 );
 
 export const setOperationCollapsed = createAction(
-    '[Canvas] Set Operation Collapsed',
+    `${CANVAS_PREFIX} Set Operation Collapsed`,
     props<{ operationCollapsed: boolean }>()
 );
 
 /*
     General
  */
 
-export const showOkDialog = createAction('[Canvas] Show Ok Dialog', props<{ title: string; message: string }>());
+export const showOkDialog = createAction(
+    `${CANVAS_PREFIX} Show Ok Dialog`,
+    props<{ title: string; message: string }>()
+);
 
 export const renderConnectionsForComponent = createAction(
-    '[Canvas] Render Connections For Component',
+    `${CANVAS_PREFIX} Render Connections For Component`,
     props<{ id: string; updatePath: boolean; updateLabel: boolean }>()
 );
 
 export const navigateToProvenanceForComponent = createAction(
-    '[Canvas] Navigate To Provenance For Component',
+    `${CANVAS_PREFIX} Navigate To Provenance For Component`,
     props<{ id: string }>()
 );
 
 export const replayLastProvenanceEvent = createAction(
-    '[Canvas] Replay Last Provenance Event',
+    `${CANVAS_PREFIX} Replay Last Provenance Event`,
     props<{ request: ReplayLastProvenanceEventRequest }>()
 );
+
+export const runOnce = createAction(`${CANVAS_PREFIX} Run Once`, props<{ request: RunOnceRequest }>());
+
+export const runOnceSuccess = createAction(`${CANVAS_PREFIX} Run Once Success`, props<{ response: RunOnceResponse }>());
+
+export const startComponent = createAction(
+    `${CANVAS_PREFIX} Start Component`,
+    props<{ request: StartComponentRequest | StartProcessGroupRequest }>()
+);
+
+export const startComponents = createAction(
+    `${CANVAS_PREFIX} Start Components`,
+    props<{ request: StartComponentsRequest }>()
+);
+
+export const startComponentSuccess = createAction(
+    `${CANVAS_PREFIX} Start Component Success`,
+    props<{ response: StartComponentResponse }>()
+);
+
+export const stopComponent = createAction(
+    `${CANVAS_PREFIX} Stop Component`,
+    props<{ request: StopComponentRequest | StopProcessGroupRequest }>()
+);
+
+export const stopComponents = createAction(
+    `${CANVAS_PREFIX} Stop Components`,
+    props<{ request: StopComponentsRequest }>()
+);
+
+export const stopComponentSuccess = createAction(
+    `${CANVAS_PREFIX} Stop Component Success`,
+    props<{ response: StopComponentResponse }>()
+);
+
+export const startCurrentProcessGroup = createAction(`${CANVAS_PREFIX} Start Current Process Group`);
+
+export const stopCurrentProcessGroup = createAction(`${CANVAS_PREFIX} Stop Current Process Group`);
+
+export const reloadProcessGroup = createAction(
```

Comment:
We have another use case where we've performed some action on a Connection and once the action is completed we need to reload the source/destination component. I think it seems like a pretty similar scenario as we've performed some action and now we want to reload a Process Group. Can these actions (`reloadProcessGroup` and `reloadProcessGroupSuccess`) be renamed and moved so they are co-located with the others? Specifically look for `loadProcessor`, `loadInputPort`, `loadRemoteProcessor`, etc. in this file.

Reply:
@mcgilman, I can certain relocate where these are defined. However, there is already an action for `loadProcessGroup` that has an effect that does much more than is needed/wanted here. It actually gets the flow, the flow status, the cluster summary, and controller bulletins and eventually navigates to the pg in the canvas.
That was why I chose to name it the way I did.